pkg/controller/kibanaassociation/association_controller.go

```diff
@@ -403,43 +401,10 @@ func (r *ReconcileAssociation) reconcileElasticsearchCA(ctx context.Context, kib
 
 // deleteOrphanedResources deletes resources created by this association that are left over from previous reconciliation
 // attempts. Common use case is an Elasticsearch reference in Kibana spec that was removed.
-func deleteOrphanedResources(ctx context.Context, c k8s.Client, kibana *kbv1.Kibana) error {
-	span, _ := apm.StartSpan(ctx, "delete_orphaned_resources", tracing.SpanTypeApp)
-	defer span.End()
-
-	var secrets corev1.SecretList
-	ns := client.InNamespace(kibana.Namespace)
-	matchLabels := NewResourceSelector(kibana.Name)
-	if err := c.List(&secrets, ns, matchLabels); err != nil {
-		return err
-	}
-
-	// Namespace in reference can be empty, in that case we compare it with the namespace of Kibana
-	var esRefNamespace string
-	if kibana.Spec.ElasticsearchRef.IsDefined() && kibana.Spec.ElasticsearchRef.Namespace != "" {
-		esRefNamespace = kibana.Spec.ElasticsearchRef.Namespace
-	} else {
-		esRefNamespace = kibana.Namespace
-	}
-
-	for _, s := range secrets.Items {
-		if metav1.IsControlledBy(&s, kibana) || hasBeenCreatedBy(&s, kibana) {
-			if !kibana.Spec.ElasticsearchRef.IsDefined() {
-				// look for association secrets owned by this kibana instance
-				// which should not exist since no ES referenced in the spec
-				log.Info("Deleting secret", "namespace", s.Namespace, "secret_name", s.Name, "kibana_name", kibana.Name)
-				if err := c.Delete(&s); err != nil && !apierrors.IsNotFound(err) {
-					return err
-				}
-			} else if value, ok := s.Labels[common.TypeLabelName]; ok && value == user.UserType &&
-				esRefNamespace != s.Namespace {
-				// User secret may live in an other namespace, check if it has changed
-				log.Info("Deleting secret", "namespace", s.Namespace, "secretname", s.Name, "kibana_name", kibana.Name)
-				if err := c.Delete(&s); err != nil && !apierrors.IsNotFound(err) {
-					return err
-				}
-			}
-		}
-	}
-	return nil
+func deleteOrphanedResources(
+	ctx context.Context,
+	c k8s.Client,
+	kibana *kbv1.Kibana,
+) error {
+	return association.DeleteOrphanedResources(ctx, c, kibana, NewResourceSelector(kibana.Name), createdByKibana)
```

Comment:
IIUC `NewResourceSelector ` selects based on `AssociationLabelName` whereas `createdByKibana ` ultimately selects based on `AssociationLabelName` + `AssociationLabelNamespace`. It feels a bit redundant and not exactly clear to me what is the different role of each.